src/toil/wdl/wdltoil.py

```diff
@@ -735,15 +748,20 @@ def evaluate_decl(node: WDL.Tree.Decl, environment: WDLBindings, stdlib: WDL.Std
 
     return evaluate_named_expression(node, node.name, node.type, node.expr, environment, stdlib)
 
-def evaluate_call_inputs(context: Union[WDL.Error.SourceNode, WDL.Error.SourcePosition], expressions: Dict[str, WDL.Expr.Base], environment: WDLBindings, stdlib: WDL.StdLib.Base) -> WDLBindings:
+def evaluate_call_inputs(context: Union[WDL.Error.SourceNode, WDL.Error.SourcePosition], expressions: Dict[str, WDL.Expr.Base], environment: WDLBindings, stdlib: WDL.StdLib.Base, inputs_list: Optional[List[WDL.Tree.Decl]] = None) -> WDLBindings:
```

Comment:
I think instead of taking `inputs_list`, we should take what's being stored in `inputs_dict` (a mapping from string to type). That makes more sense alongside `expressions`, which is a mapping from string to expression, and keeps `WDL.Tree` out of the function. We should push the code to convert from the list of `Decl`s to a collection of expected types out into the caller.

The docstring should explain what the new parameter is and how it needs to relate to `expressions`, and that is a lot easier if we can say it is "an optional dict from name to expected type, for the names in `expressions`", versus if we have to say it is "An optional list of WDL Decl workflow nodes, where there is a Decl for each entry in `expressions`, and the name of the Decl is the same as the name used for the expression, and the type of the Decl is the expected type for the expression".

It also would make it a lot easier to write a test for, since it's harder to construct `WDL.Tree.Decl` objects by hand.